src/Autodesk.Forge.DesignAutomation/Model/ArgumentConverter.cs

```diff
@@ -33,25 +33,34 @@ public override IArgument Create(Type objectType)
 
         public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
         {
+            IArgument target;
             if (reader.TokenType == JsonToken.StartObject)
             {
-                IArgument target;
-                var jObject = JObject.Load(reader);
-                if (jObject.Property("url") != null)
+                JObject jObject = JObject.Load(reader);
+                if (jObject["url"] != null)
                 {
                     target = new XrefTreeArgument();
+                    serializer.Populate(jObject.CreateReader(), target);
+                }
+                else if (jObject["value"] != null)
+                {
+                    target = new StringArgument(jObject["value"].Value<string>());
                 }
                 else
                 {
-                    target = new StringArgument();
+                    throw new JsonSerializationException($"Expected XrefTreeArgument or StringArgument.");
                 }
-                serializer.Populate(jObject.CreateReader(), target);
-                return target;
+            }
+            else if (reader.TokenType == JsonToken.String)
+            {
+                target = new StringArgument(reader.Value.ToString());
             }
             else
             {
-                throw new JsonReaderException("Expected start of object.");
+                throw new JsonSerializationException($"Expected object or string but got {reader.TokenType}.");
```

Comment:
Why are you changing the exception type?

Reply:
Do we have some test that hit this error case?

Reply:
I just _copied_ the latest ReadJson code. Does the change break something? If so, I can revert it back.
I don't think we have e2e tests to hit the error case, should we add some?

Reply:
> Does the change break something?

I don't know that's why I'm asking. Yes, we need a test that hits this code so that we know whether we handle it correctly.

Reply:
Ok